zebrad/src/commands/start.rs

```diff
@@ -54,6 +54,27 @@ impl StartCmd {
             config.network.network,
         ));
 
+        info!("starting legacy chain check");
+        match state
+            .clone()
+            .oneshot(zebra_state::Request::IsLegacyChain)
+            .await
+        {
+            Ok(zebra_state::Response::LegacyChain(response)) => {
+                if response.is_some() {
+                    // we found a legacy chain
+                    panic!(
+                        "Legacy chain found, database can be corrupted.
+                        Delete your database and retry a full sync.
+                        Database path: {:?}",
+                        response.expect("the corrupted database path")
+                    );
```

Comment:
Minor styling: you could avoid the `expect` with something like:

```suggestion
                if let Some(corrupt_db_path) = response {
                    // we found a legacy chain
                    panic!(
                        "Legacy chain found, database can be corrupted.
                        Delete your database and retry a full sync.
                        Database path: {:?}",
                        corrupt_db_path,
                    );
```

Reply:
Thanks, i added this one in https://github.com/ZcashFoundation/zebra/pull/2366/commits/796fc7c558cd48376b7a34b9775b3f3545b9931f